flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/runner/JobDispatcherLeaderProcessFactoryFactory.java

```diff
@@ -131,8 +131,12 @@ private static Optional<JobGraph> getJobGraphBasedOnDirtyJobResults(
             JobGraph jobGraph, Collection<JobResult> dirtyJobResults) {
         final Set<JobID> jobIdsOfFinishedJobs =
                 dirtyJobResults.stream().map(JobResult::getJobId).collect(Collectors.toSet());
-        return jobIdsOfFinishedJobs.contains(jobGraph.getJobID())
-                ? Optional.empty()
-                : Optional.of(jobGraph);
+        if (jobIdsOfFinishedJobs.contains(jobGraph.getJobID())) {
+            LOG.info(
```

Comment:
put inside `getJobGraphBasedOnDirtyJobResults` since there is also `JobDispatcherLeaderProcessFactoryFactory#extractDirtyJobResult` doing some logging and probably it would make sense to have both doing it in corresponding methods or both in `createFactory`